apache/superset
# fix: Popover closes on change of dropdowns values

### SUMMARY
The option `menuPortalTarget` of the react-select component was introduced in order for the dropdowns to show fully when overlaying certain container elements. However, setting the `menuPortalTarget` to `document.body` (as originally the multi-choice dropdown did and later the single-choice) comes with several issues, such as the click events being targeted outside of the container elements, that ultimately caused this issue #12388.

This PR removes the `menuPortalTarget` for both single-choice and multi-choice dropdowns. In addition to that, this introduces the optional ability to force the overflow visibility of the dropdowns for cases where setting the `overflow` `visible` isn't viable on the container element (see for instance issue #12004 )

It also adjusts some excessive z-index by setting it high enough to give it precedence over the Antd Popover and other components.

Closes #12388 

### BEFORE

https://user-images.githubusercontent.com/60598000/104368923-06b6fc00-551d-11eb-935b-b6b7ba657dc3.mp4

### AFTER

https://user-images.githubusercontent.com/60598000/104368799-dc653e80-551c-11eb-9778-3fed7cede574.mp4



### TEST PLAN
1. Open the single and multi-choice dropdowns throughout the project
2. Make sure the dropdowns are displaying correctly

### ADDITIONAL INFORMATION


- [x] Has associated issue: #12388
